yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -657,6 +657,9 @@ private[spark] class Client(
           case e: ApplicationNotFoundException =>
             logError(s"Application $appId not found.")
             return (YarnApplicationState.KILLED, FinalApplicationStatus.KILLED)
+          case ie: IOException =>
+            logError(s"Application $appId occurred an unexpected IOException.")
+            return (YarnApplicationState.KILLED, FinalApplicationStatus.KILLED)
```

Comment:
So this fixes this particular case, which is fine, but it seems there's a deeper underlying issue here: any exception thrown by this method will case the process to "hang".

So, shouldn't this just catch any exception instead? Then log it with a generic message (`logError("Exception while monitoring application", e)`) and be done with it.